compiler/rustc_session/src/options.rs

```diff
@@ -1163,6 +1163,8 @@ options! {
         "the codegen unit partitioning strategy to use"),
     chalk: bool = (false, parse_bool, [TRACKED],
         "enable the experimental Chalk-based trait solving engine"),
+    chalk_migration: bool = (false, parse_bool, [TRACKED],
+        "compare current trait solving engine against the experimental Chalk-based one"),
```

Comment:
I think `-Zchalk-migration` should either *require* `-Zchalk` or *imply* it. Particularly because there are already a couple places where we check if we have `-Zchalk` and do something a little different (like elaborate wf types in the env).

Reply:
Yea, I was wondering about that... i'll do some more experiments

Reply:
hmm... neither passing -Zchalk nor not passing it will give us exactly what we want. Either we brick legacy mode due to passing `-Zchalk` or we brick chalk mode by not passing it. Not sure how to best approach this.

Reply:
Probably need to thread through some sort of `chalk` flag on `InferCtxt` or something, rather than relying on `sess`

Reply:
Sounds more hacky than just adding a flag on InferCtxt

Reply:
Hmm. Maybe the better course is to figure out exactly where the behavior differs (outside fulfillment) and try to reduce that to 0? Or...can we even.

It might just be that now is "not the right time" for this PR.

Reply:
Yea, I did look into that already and you're right. That's not feasible right now